File: Day1/Problem1-CodeChef-CVDRUN/sol1.py
t = int(input())
while t:
  t= t-1
  n,k,x,y = [int(l) for l in input().split()]
  count = [0]*n
  while True:
    if count[x]:
      flag = 1
      break
    count[x] +=1
    x = (x+k)%n
    if count[y]:
      flag = 2
      break
  if k==0 or k==n:
    if x==y:
      print("YES")
    else:
      print("NO")
  elif flag==2:
    print("YES")
  else:
    print("NO")

File: Day2/Problem1-CodeChef-CHEFEZQ/sol1.cpp
#include <iostream>
using namespace std;

#define ll long long

int main() {
	ll T;
	cin>>T;
	while(T--) {
	  ll n,k, count = 0, flag = 0;
	  cin>>n>>k;
	  ll arr[n];
	  for(ll i=0; i<n; i++) {
	    cin>>arr[i];
	  }
	  for(ll i=0; i<n-1; i++){
	    if(arr[i] >= k) {
	      arr[i+1] += arr[i] - k;
	      count++;
	    } else {
	      count++;
	      flag = 1;
	      break;
	    }
	  }
	  if(flag == 1) 
	    cout<<count<<endl;
	  else
	    cout<<count + 1 + arr[n-1]/k<<endl;
	}
	return 0;
}
